### PR TITLE
329 multicolour accordion

### PR DESCRIPTION
**Pull request by sonaldhekale**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--cn-website--netcentric.hlx.page/
- After: https://329-multicolour-accordion--cn-website--netcentric.hlx.page/drafts/sdhekale/multicolor-accordian

### CODE REVIEW

File: blocks/multicolor-accordion/multicolor-accordion.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+export default async function decorate(block) {
+  const accordionItems = block.querySelectorAll(':scope > div > div');
+  accordionItems.forEach((accordionItem) => {
+    const nodes = accordionItem.children;
+
+    const titleText = nodes[0];
+    const rest = Array.prototype.slice.call(nodes, 1);
```
Comment (nc-andreashaller) [inline blocks/multicolor-accordion/multicolor-accordion.js:7]: why not calling slice directly? `nodes.slice(1)`?

File: blocks/multicolor-accordion/multicolor-accordion.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    const titleDiv = document.createElement('div');
+    const detailsBtn = document.createElement('button');
+    detailsBtn.innerText = 'Details';
```
Comment (nc-andreashaller) [inline blocks/multicolor-accordion/multicolor-accordion.js:11]: we need this text at least configurable via placeholder if not overwrite-able in the block

File: blocks/multicolor-accordion/multicolor-accordion.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+export default async function decorate(block) {
+  const accordionItems = block.querySelectorAll(':scope > div > div');
+  accordionItems.forEach((accordionItem) => {
+    const nodes = accordionItem.children;
+
+    const titleText = nodes[0];
```
Comment (nc-andreashaller) [inline blocks/multicolor-accordion/multicolor-accordion.js:6]: taking the first line feels a bit random, shouldn't we use a headline for marking the title?

File: blocks/multicolor-accordion/multicolor-accordion.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const newItem = document.createElement('div');
+    newItem.appendChild(titleDiv);
+    newItem.appendChild(content);
+
+    newItem.classList.add('accordion-item');
+
+    accordionItem.replaceWith(newItem);
+  });
+
+  const triggers = block.querySelectorAll('.accordion-trigger');
```
Comment (nc-andreashaller) [inline blocks/multicolor-accordion/multicolor-accordion.js:32]: we are just selecting the titleDiv here which we have already above, makes probably sense to directly attach the listener above

File: blocks/multicolor-accordion/multicolor-accordion.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  triggers.forEach((trigger) => {
+    trigger.addEventListener('click', () => {
+      const openAttribute = 'aria-expanded';
+      const wasOpen = trigger.parentElement.hasAttribute(openAttribute);
+
+      triggers.forEach((_trigger) => {
+        const getDetailsBtn = _trigger.children[1];
```
Comment (nc-andreashaller) [inline blocks/multicolor-accordion/multicolor-accordion.js:39]: what are we doing here with the button? I don't get this block of code

**Review by nc-andreashaller — CHANGES_REQUESTED** [review]
see comments

**Comment by sonaldhekale** [comment]
hi changes requested are done have a look @nc-andreashaller.Thanks

File: blocks/multicolor-accordion/multicolor-accordion.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+.multicolor-accordion>div:nth-child(odd) { 
+  background: var(--c-dark-plum);
+}
+
+.multicolor-accordion>div:nth-child(even) { 
+    background: var(--c-dark-blue);
+  }
+
+  .accordion-trigger .btn {
```
Comment (nc-andreashaller) [inline blocks/multicolor-accordion/multicolor-accordion.css:25]: formatting seems a little broken here, please check indentation

File: blocks/multicolor-accordion/multicolor-accordion.js (hunk 3 above)
Comment (nc-andreashaller) [reply blocks/multicolor-accordion/multicolor-accordion.js:20]: did you just close this without change?

**Review by nc-andreashaller — DISMISSED** [review]
As discussed via Slack please review this PR together in a pair programming style with an experienced developer and let me know once it is ready to be reviewed.

File: blocks/multicolor-accordion/multicolor-accordion.js (hunk 3 above)
Comment (sonaldhekale) [reply blocks/multicolor-accordion/multicolor-accordion.js:20]: now its following column wise structure so node[0] is the first column its a title.

**Review by mihai-buricea-net — APPROVED** [review]

**Comment by alexnerdrumhansen** [comment]
A couple of concerns. 
1. We need to make a more mobile friendly version when viewed on smaller viewports. @Gonzalo-CG Can you have a look at this please? Currently only a short text displays on mobile and we'd want a good experience for those visiting from mobile too.
2. I tried to add a link in one of the drawers to verify that works, but once I edited the accordion no longer shows properly on the page. Is there a trick to getting it display as an accordion after editing @sonaldhekale ?

Thanks!

Attaching mobile view
Screenshot 2023-08-08 at 09 23 09

**Comment by Gonzalo-CG** [comment]
@sonaldhekale @alexnerdrumhansen as for the mobile view, Joel recommended reducing the font size & eliminating the padding on the left side.

**Comment by alexnerdrumhansen** [comment]
Changes made make it look much better on mobile.
Can now also verify that links will open in new tabs.

Thanks @sonaldhekale !
Screenshot 2023-08-09 at 15 03 22

**Pull request merged**